src/utils.cpp

```diff
@@ -31,11 +31,11 @@
 #include <zypp/ZYppFactory.h>
 #include <zypp/ResPool.h>
 #include <zypp/PoolItemBest.h>
+#include <zypp/base/Easy.h>
 #include "utils.h"
 
 // Define static variables here
-RepoManagerOptions* ZypperUtils::s_repoManagerOpts = new RepoManagerOptions("/tmp");
-RepoManager* ZypperUtils::s_repoManager = new RepoManager(*s_repoManagerOpts);
+scoped_ptr<RepoManager> ZypperUtils::s_repoManager( new RepoManager( Pathname( "/tmp" ) ) );
```

Comment:
I concur. I should probably use `/tmp/oci-RANDOM`

The idea here is to display meta-information of packages to the user. Default constructor `(new RepoManager())` needs root privileges which is rather much too early to call xdg-su on `OCI-helper`. Hence, loading the meta-information to `/tmp` (`/tmp/oci-RANDOM` from now on) to display the information. 

When the user clicks on "Install Software" button (check for links below), [after taking care of other stuff] call to `makeNecessaryChangesToRunAsRoot()` is made to reset the `s_repoManager` to add specific repo to the system repos (OCI runs akin to  `zypper in <package>` from this point onward ).

Strictly speaking, when "Install Software" button is clicked, OCI calls `OCI-helper` with root privileges. `OCI-helper` takes care of adding repos to the system, installing packages.. and so on and on. ;)

Here's the intended work(-ish) flow of OCI
https://cloud.githubusercontent.com/assets/3799140/13884200/99e6a298-ecf2-11e5-87fa-4e9efcb22ffa.png
https://raw.githubusercontent.com/wiki/openSUSE/one-click-installer/oneclick-mockup.png